### PR TITLE
CMSIS-NN: Add dilation support for generic depthwise conv

### PR DESCRIPTION
**Pull request by mansnils**
Adds dilation support for depthwise conv operator.

### CODE REVIEW

**Comment by grasci-arm** [comment]
Can one of the admins verify this patch?

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_depthwise_conv_s8/test_arm_depthwise_conv_s8.c [hunk 1]
```diff
@@ -558,4 +579,67 @@ void depthwise_null_bias_1_arm_depthwise_conv_s8(void)
...
+    dw_conv_params.output_offset = DEPTHWISE_DILATION_OUTPUT_OFFSET;
+    dw_conv_params.activation.min = DEPTHWISE_DILATION_OUT_ACTIVATION_MIN;
+    dw_conv_params.activation.max = DEPTHWISE_DILATION_OUT_ACTIVATION_MAX;
+    quant_params.multiplier = (int32_t *)depthwise_dilation_output_mult;
+    quant_params.shift = (int32_t *)depthwise_dilation_output_shift;
+
+    ctx.buf = NULL;
+    ctx.size = 0;
+
+    arm_status result = arm_depthwise_conv_s8(&ctx,
```
Comment (felix-johnny) [inline CMSIS/NN/Tests/UnitTest/TestCases/test_arm_depthwise_conv_s8/test_arm_depthwise_conv_s8.c:630]: Lets call the wrapper here so the entire chain is verified.

File: ARM.CMSIS.pdsc [hunk 2]
```diff
@@ -15,6 +15,7 @@
        - Support for int16 convolution and fully connected for reference implementation
        - Support for DSP extension optimization for int16 convolution and fully connected
        - Support for dilation for int8 convolution
+       - Support for dilation for int8 depthwise convolution
```
Comment (felix-johnny) [inline ARM.CMSIS.pdsc:18]: Please rephrase it as 'Support dilation for int8 depthwise convolution'. The same for Convolution

File: CMSIS/NN/Tests/UnitTest/TestCases/test_arm_depthwise_conv_s8/test_arm_depthwise_conv_s8.c (hunk 1 above)
Comment (mansnils) [reply CMSIS/NN/Tests/UnitTest/TestCases/test_arm_depthwise_conv_s8/test_arm_depthwise_conv_s8.c:630]: Done

File: ARM.CMSIS.pdsc (hunk 2 above)
Comment (mansnils) [reply ARM.CMSIS.pdsc]: Done

**Review by felix-johnny — APPROVED** [review]

**Pull request merged**